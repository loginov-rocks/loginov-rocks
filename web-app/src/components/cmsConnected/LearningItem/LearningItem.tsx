import * as React from 'react';

import { CmsComponent } from 'cms/interfaces/CmsComponent';
import { CmsConnectedProps } from 'cms/interfaces/CmsConnectedProps';
import { ExternalLink } from 'components/shared/ExternalLink';
import { Time } from 'components/shared/Time';

interface Props extends CmsConnectedProps {
  completed?: string;
  credentialId?: string;
  credentialLink?: string;
  provider?: CmsComponent;
  providerLink?: string;
  timeToComplete?: number;
  title: string;
}

export const LearningItem: React.FC<Props> = ({
  completed, credentialId, credentialLink, provider, providerLink, render, timeToComplete, title,
}) => {
  let renderTitle = <strong>{title}</strong>;

  if (providerLink) {
    renderTitle = <ExternalLink href={providerLink}>{renderTitle}</ExternalLink>;
  }

  let renderCredential = null;

  if (credentialId && credentialLink) {
    renderCredential = <ExternalLink href={credentialLink}>{credentialId}</ExternalLink>;
  } else if (credentialId) {
    renderCredential = credentialId;
  } else if (credentialLink) {
    renderCredential = <ExternalLink href={credentialLink}>Link</ExternalLink>;
  }

  return (
    <>

      {renderTitle}

      {provider && (
        <>
          {' '}
          /
          {' '}
          {render(provider)}
        </>
      )}

      {(completed || timeToComplete) && (
        <>
          <br />

          {completed && (
            <span title="Completed">
              {new Date(completed).toLocaleDateString(undefined, {
                day: 'numeric',
                month: 'short',
                year: 'numeric',
              })}
            </span>
          )}

          {completed && timeToComplete && (
            <>
              {' '}
              /
              {' '}
            </>
          )}

          {timeToComplete && (
            <Time label="Time to Complete" minutes={timeToComplete} />
          )}

        </>
      )}

      {renderCredential && (
        <em>
          <br />
          Credential:
          {' '}
          {renderCredential}
        </em>
      )}

    </>
  );
};

LearningItem.defaultProps = {
  completed: undefined,
  credentialId: undefined,
  credentialLink: undefined,
  provider: undefined,
  providerLink: undefined,
  timeToComplete: undefined,
};
