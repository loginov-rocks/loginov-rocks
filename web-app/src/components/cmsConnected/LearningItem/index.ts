import { LearningItem } from './LearningItem';

export { LearningItem };
