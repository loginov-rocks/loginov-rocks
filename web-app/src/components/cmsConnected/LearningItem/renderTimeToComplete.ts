export const renderTimeToComplete = (minutes: number): string => {
  const hours = Math.floor(minutes / 60);
  const remainingMinutes = minutes % 60;

  let string = '';

  if (hours > 0) {
    string += `${hours} hour${hours !== 1 ? 's' : ''}`;
  }

  if (remainingMinutes > 0) {
    if (string.length > 0) {
      string += ', ';
    }

    string += `${remainingMinutes} minute${remainingMinutes !== 1 ? 's' : ''}`;
  }

  return string;
};
