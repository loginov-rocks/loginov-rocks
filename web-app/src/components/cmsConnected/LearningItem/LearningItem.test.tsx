/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import { LearningItem } from './LearningItem';

it('matches snapshot', () => {
  const wrapper = shallow(
    <LearningItem
      completed="2018-08-05"
      credentialId="AdMN2oSrbxB6Uv1mNo9hpTeLY9s0"
      // eslint-disable-next-line max-len
      credentialLink="https://www.linkedin.com/learning/certificates/79a9ebf220d4c1dd2ebc94508415949a21614443b37b7e2fc02de3a1a164d420"
      provider={{
        props: {
          title: 'LinkedIn Learning',
          website: 'https://www.linkedin.com/learning/',
        },
        type: 'learningProvider',
      }}
      providerLink="https://www.linkedin.com/learning/learning-web-components"
      render={() => null}
      timeToComplete={84}
      title="Learning Web Components"
    />,
  );

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot with optional fields', () => {
  const wrapper = shallow(<LearningItem render={() => null} title="Learning Web Components" />);

  expect(wrapper).toMatchSnapshot();
});
