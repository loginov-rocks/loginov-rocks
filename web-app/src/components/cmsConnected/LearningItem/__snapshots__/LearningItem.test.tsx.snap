// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <ExternalLink
    href="https://www.linkedin.com/learning/learning-web-components"
    label="Link to Certification or Training"
  >
    <strong>
      Learning Web Components
    </strong>
  </ExternalLink>
   
  /
   
  <br />
  <span
    title="Completed"
  >
    Aug 5, 2018
  </span>
   
  /
   
  <Time
    label="Time to Complete"
    minutes={84}
  />
  <em>
    <br />
    Credential:
     
    <ExternalLink
      href="https://www.linkedin.com/learning/certificates/79a9ebf220d4c1dd2ebc94508415949a21614443b37b7e2fc02de3a1a164d420"
      label="Link to Credential"
    >
      AdMN2oSrbxB6Uv1mNo9hpTeLY9s0
    </ExternalLink>
  </em>
</Fragment>
`;

exports[`matches snapshot with optional fields 1`] = `
<Fragment>
  <strong>
    Learning Web Components
  </strong>
</Fragment>
`;
